chisel/src/solidity_helper.rs

```diff
@@ -158,13 +157,7 @@ impl SolidityHelper {
         let mut errors = Vec::with_capacity(1);
         for res in Lexer::new(input, 0, &mut comments, &mut errors) {
             match res {
-                Err(err) => match err {
-                    LexicalError::EndOfFileInComment(_) |
-                    LexicalError::EndofFileInHex(_) |
-                    LexicalError::EndOfFileInString(_) => return ValidationResult::Incomplete,
-                    _ => return ValidationResult::Valid(None),
-                },
-                Ok((_, token, _)) => match token {
+                (_, token, _) => match token {
```

Comment:
can replace `res` in the `for` above with this pattern and remove the match entirely

```
for (_, token, _) in ...
```